### ISSUE TITLE
Binding title on ActionBar removes custom NavigationButton placed inside.

### ISSUE THREAD

**Issue by manojdcoder** [NativeScript/nativescript-angular#906]
When binding `title` of `ActionBar`, the custom `NavigationButton` placed inside is replaced by default one.

```
<ActionBar [title]="title" class="action-bar">
    <NavigationButton android.systemIcon="ic_menu_back" (tap)="onBackButtonTap()">
    </NavigationButton>
</ActionBar>
```

The moment a value is assigned to the variable `title` may be on `ngOnInit`, the custom `NavigationButton` button placed with a listener is removed / reset. `onBackButtonTap` is never called. If we don't bind `title` listener is called as expected.

**Versions:**
TNS CLI: 3.1.2
Runtime & Core Modules: 3.0.1
Angular: 4.1.0
Platform: iOS

**Comment by tsonevn** [comment]
Hi @manojdcoder,
I tested this behavior with latest `tns-core-modules` and `nativescript-angular` 3.1.0 and indeed there is an issue with firing the tap event for  `NavigationButton` on iOS. 
I am attaching the project, where the problem could be reproduced. You could keep track on the issue for further info.

In the meantime, you could use `ActionItem` with `ios.position="left"`. For example:
```
<ActionItem (tap)="onShare()"
      ios.systemIcon="9" ios.position="left"
      android.systemIcon="ic_menu_share" 
      android.position="actionBar"></ActionItem>
```
[Archive.zip](https://github.com/NativeScript/nativescript-angular/files/1158840/Archive.zip)

**Comment by manojdcoder** [comment]
@tsonevn Thank you for the quick response and work around.

**Comment by manojdcoder** [comment]
@tsonevn I tried the workaround you suggested, still the navigation button is automatically placed in left side along with the left action item. Any idea how we can prevent that?

**Comment by manojdcoder** [comment]
@tsonevn I managed to hide the `NavigationButton` by setting `visibility` property. Guess it not only has problem with firing `tap` event but also other properties like `text`. Whatever `text` we set, then it gets reset to `Back`.

**Comment by tsonevn** [comment]
Hi @manojdcoder,
This problem should be fixed in @next nativescript-angular, could you verify whether the problem will still exist, while using @next. 
Steps:
`tns plugin remove nativescript-angular`
`tns plugin add nativescript-angular@next`

Bear in mind to delete `node_modules` and `platforms` folders, before rebuilding.

**Comment by manojdcoder** [comment]
@tsonevn I'm getting the error below when running with next.

```
node_modules/nativescript-angular/router/page-router-outlet.d.ts(2,26): error TS2305: Module '"ns-angular-906/node_modules/@angular/router/router"' has no exported member 'ChildrenOutletContexts'.

Unable to apply changes on device: F2029514-A16E-4A95-B17D-AC11B2BB0278. Error is: TypeScript compiler failed with exit code 1.
```

**Comment by tsonevn** [comment]
@manojdcoder,
You should also change you `@angular` dependencies version to 4.3.3 as it is shown in the below-attached sample `package.json` file:

```
{
  "description": "NativeScript Application",
  "license": "SEE LICENSE IN <your-license-filename>",
  "readme": "NativeScript Application",
  "repository": "<fill-your-repository-here>",
  "nativescript": {
    "id": "org.nativescript.test",
    "tns-android": {
      "version": "3.1.1"
    }
  },
  "dependencies": {
    "@angular/animations": "~4.1.0",
    "@angular/common": "~4.3.3",
    "@angular/compiler": "~4.3.3",
    "@angular/core": "~4.3.3",
    "@angular/forms": "~4.3.3",
    "@angular/http": "~4.3.3",
    "@angular/platform-browser": "~4.3.3",
    "@angular/router": "~4.3.3",
    "nativescript-angular": "^3.2.0-2017-8-8-5",
    "nativescript-theme-core": "~1.0.2",
    "reflect-metadata": "~0.1.8",
    "rxjs": "~5.3.0",
    "tns-core-modules": "3.1.1",
    "zone.js": "~0.8.2"
  },
  "devDependencies": {
    "babel-traverse": "6.25.0",
    "babel-types": "6.25.0",
    "babylon": "6.17.4",
    "lazy": "1.0.11",
    "nativescript-dev-typescript": "~0.5.0",
    "typescript": "~2.2.1"
  }
}
```

**Comment by manojdcoder** [comment]
@tsonevn The issue doesn't seem fixed yet, I'm trying the same example you had attached in your first comment. No alert neither log, just moves back to previous page.

**Comment by tsonevn** [comment]
Hi @manojdcoder,
Thank you for checking.
I will keep researching the problem and will provide more info about the case.

**Comment by tsonevn** [comment]
Hi @manojdcoder,
Excuse me for the misunderstanding, 
In my further research, I found that this is expected behavior for iOS and the tap event will not be fired for the NavigationBack button. This is described also in the documentation [here](https://docs.nativescript.org/ui/action-bar#ios-specifics). The solution, in this case, is to setup `ActionItem` with a left position as I suggested at the beginning of this thread.

Regarding that, I am closing this issue.
